Sources/PackageLoading/TargetSourcesBuilder.swift

```diff
@@ -454,6 +476,49 @@ public struct TargetSourcesBuilder {
 
         return contents
     }
+
+    public static func computeContents(for generatedFiles: [AbsolutePath], toolsVersion: ToolsVersion, additionalFileRules: [FileRuleDescription], defaultLocalization: String?, targetName: String, targetPath: AbsolutePath, observabilityScope: ObservabilityScope) -> (sources: [AbsolutePath], resources: [Resource]) {
+        var sources = [AbsolutePath]()
+        var resources = [Resource]()
+
+        generatedFiles.forEach { absPath in
+            // 5.6 handled treated all generated files as sources.
+            if toolsVersion <= .v5_6 {
+                sources.append(absPath)
+                return
+            }
+
+            var rule = Self.computeRule(for: absPath, toolsVersion: toolsVersion, additionalFileRules: additionalFileRules, observabilityScope: observabilityScope)
+
+            // If we did not find a rule for a generated file, we treat it as to be processed for now. Eventually, we should handle generated files the same as other files and require explicit handling in the manifest for unknown types.
+            if rule == .none {
+                rule = .processResource(localization: .none)
+            }
```

Comment:
we can also make .none a case in the switch below instead of explicitly handling it here

Reply:
We could, but I think it's easier to understand this way this way than doing any weird dance with `fallthrough` in the `switch` to avoid duplication.

Reply:
would it not just be `case .none, .copy, .processResource:`?

Reply:
But we have to swap the rule to the right one before doing the code for copy/process.

Reply:
I guess we could do something like

```
let actualRule = rule == .none ? .processResource(localization: .none) : rule
```

or pull the `if` statement into the `switch`, but I think that's more difficult to understand.

Reply:
I missed the fact we need to send the rule into `Self.resource`. we could tease `.none` and `.processResource` case handling apart from `.copy`and pass in `.processResource` into `Self.resource` for both. that would feel the cleanest imo.

in any case its not a blocker for this PR

Reply:
I think I prefer the current structure since it calls more attention to the `.none` handling being a special case before we get to the regular handling of file types, instead of merging the two.